src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4613,7 +4613,11 @@ private static bool IsTargetTypedExpression(BoundExpression node)
             {
                 var (returnExpr, resultType, _) = returns[i];
                 resultTypes.Add(resultType);
-                placeholdersBuilder.Add(CreatePlaceholderIfNecessary(returnExpr, resultType));
+
+                if (!IsTargetTypedExpression(returnExpr))
```

Comment:
>if (!IsTargetTypedExpression(returnExpr)) [](http://example.com/codeflow?start=16&length=41)

Is this change covered by a test? It didn't seem like the new tests were affected by this check.

Reply:
That's correct, but, I would prefer to leave this check in order to satisfy new assertions in BestTypeInferrer.

I didn't investigate closely but I am guessing something is causing the return type we wanted to infer here to be replaced unexpectedly with a type from initial binding.

Reply:
Actually since this is only used in initial binding we might be able to remove the check and just assert that a target typed expr is not being used here.

Reply:
urgh, this code probably would have behaved incorrectly if one of the exprs actually was target-typed, it expects that 'resultTypes ' and 'placeholdersBuilder' have the same count after the loop.